File: data/multilang_comparison/program_c/src/func8.c
#include "program_c.h" 
array_t* func8(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array170;
      if (pCounter > 0) {
         array170 = vars->data[--pCounter];
         array170->refC++;
         DEBUG_COPY(array170->id);
      } else {
         array170 = (array_t*)malloc(sizeof(array_t));
         array170->size = 627;
         array170->refC = 1;
         array170->id = 170;
         array170->data = (unsigned int*)malloc(array170->size*sizeof(unsigned int));
         memset(array170->data, 0, array170->size*sizeof(unsigned int));
         DEBUG_NEW(array170->id);
      }
      unsigned int loop50 = 0;
      unsigned int loopLimit50 = (50)/2 + 1;
      for(; loop50 < loopLimit50; loop50++) {
         for (int i = 0; i < array170->size; i++) {
            array170->data[i]--;
         }
         for (int i = 0; i < array170->size; i++) {
            if (array170->data[i] == 14) { 
               return array170;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array170;
      array_t* array171 = func18(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array171->id);
      free(params0.data);
      array171->refC--;
      if(array171->refC == 0) {
         free(array171->data);
         free(array171);
         DEBUG_FREE(array171->id);
      }
      array170->refC--;
      if(array170->refC == 0) {
         free(array170->data);
         free(array170);
         DEBUG_FREE(array170->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array172 = func14(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array172->id);
      free(params0.data);
      array172->refC--;
      if(array172->refC == 0) {
         free(array172->data);
         free(array172);
         DEBUG_FREE(array172->id);
      }
   }
   array_t* array173;
   if (pCounter > 0) {
      array173 = vars->data[--pCounter];
      array173->refC++;
      DEBUG_COPY(array173->id);
   } else {
      array173 = (array_t*)malloc(sizeof(array_t));
      array173->size = 479;
      array173->refC = 1;
      array173->id = 173;
      array173->data = (unsigned int*)malloc(array173->size*sizeof(unsigned int));
      memset(array173->data, 0, array173->size*sizeof(unsigned int));
      DEBUG_NEW(array173->id);
   }
   return array173;
}

File: data/multilang_comparison/program_nim/src/func8.nim
proc func8*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var array266: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array266 = vars.data[pCounter]
         inc array266.refC
         debugCopy(array266.id)
      else:
         array266 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array266.size = 925
         array266.refC = 1
         array266.id = 266
         array266.data = cast[ptr UncheckedArray[uint32]](alloc(array266.size * sizeof(uint32)))
         zeroMem(array266.data, array266.size * sizeof(uint32))
         debugNew(array266.id)
      let loopLimit77 = (50) div (2) + 1
      for loop77 in 0 ..< loopLimit77:
         discard
         for i in 0 ..< array266.size:
            dec array266.data[i]
         for i in 0 ..< array266.size:
            if array266.data[i] == 18:
               return array266
      var params0: ArrayTParam
      params0.size = 1
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array266
      var array267: ptr ArrayT = func18(addr params0, getPath(), loopsFactor)
      debugReturn(array267.id)
      dealloc(params0.data)
      dec array267.refC
      if array267.refC == 0:
         dealloc(array267.data)
         dealloc(array267)
         debugFree(array267.id)
      dec array266.refC
      if array266.refC == 0:
         dealloc(array266.data)
         dealloc(array266)
         debugFree(array266.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array268: ptr ArrayT = func14(addr params0, getPath(), loopsFactor)
      debugReturn(array268.id)
      dealloc(params0.data)
      dec array268.refC
      if array268.refC == 0:
         dealloc(array268.data)
         dealloc(array268)
         debugFree(array268.id)
   var array269: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array269 = vars.data[pCounter]
      inc array269.refC
      debugCopy(array269.id)
   else:
      array269 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array269.size = 90
      array269.refC = 1
      array269.id = 269
      array269.data = cast[ptr UncheckedArray[uint32]](alloc(array269.size * sizeof(uint32)))
      zeroMem(array269.data, array269.size * sizeof(uint32))
      debugNew(array269.id)
   return array269

